Code/GraphMol/ChemTransforms/MolFragmenter.cpp

```diff
@@ -920,8 +921,14 @@ struct ZipBond {
 };
 }  // namespace
 
-std::unique_ptr<ROMol> molzip(const ROMol &a, const ROMol &b,
-                              const MolzipParams &params) {
+static const std::string indexPropName("__zipIndex");
+
+std::unique_ptr<ROMol> molzip(
+    const ROMol &a, const ROMol &b, const MolzipParams &params,
+    std::optional<std::map<int, int>> &attachmentMapping) {
```

Comment:
huh, that's the first time `std::optional` shows up in the RDKit.

Reply:
hopefully that is not a bad thing.

Reply:
nope!

Reply:
Code/GraphMol/ChemTransforms/MolFragmenter.cpp needs to #include &lt;optional&gt; in order to define/use std::optional.  Otherwise the code fails to build with g++ 11.4.  Other compilers may have (non-portably) defined std::optional via other standard headers.

Reply:
@ricrogz, would you mind adding this to your big cleanup PR too?

Reply:
Done!